test/functional/job/job_spec.lua

```diff
@@ -104,6 +104,17 @@ describe('jobs', function()
     nvim('eval', 'jobsend(j, "abcdef")')
     nvim('eval', 'jobstop(j)')
     eq({'notification', 'j', {{jobid, 'stdout', {'abcdef'}}}}, next_message())
-    eq({'notification', 'j', {{jobid, 'exit'}}}, next_message())
+    eq({'notification', 'j', {{jobid, 'exit', 0}}}, next_message())
+  end)
+
+  it('will return the proper exit status for the job', function()
+    nvim('command', notify_job())
+    nvim('command', "let j = jobstart('xxx', 'cat', ['/dev/null/fake'])")
+    local jobid = nvim('eval', 'j')
+    next_message()
+    eq({'notification', 'j', {{jobid, 'exit', 1}}}, next_message())
+    nvim('command', "let j = jobstart('xxx', 'bash', ['-c', 'exit 100'])")
+    local jobid = nvim('eval', 'j')
+    eq({'notification', 'j', {{jobid, 'exit', 100}}}, next_message())
```

Comment:
According to [wikipedia](http://en.wikipedia.org/wiki/Exit_status) a lot of systems (including POSIX, bash) limit exit status codes to between 0 and 255. Trying values outside of this range just seems to cause integer overflow in bash. I could use `255` instead of `100` here so that both extremes are tested.